returnn/datasets/postprocessing.py

```diff
@@ -263,3 +289,29 @@ def _get_seq_len(self, tdict: TensorDict) -> int:
         :return: segment length of the segment in `tdict` as measured by `self.length_key` for comparison.
         """
         return tdict.data[self.length_key].raw_tensor.shape[0]
+
+
+T = TypeVar("T", TensorDict, Iterator[TensorDict])
+
+
+class Sequential(Generic[T]):
+    """
+    Callable that composes multiple postprocessing functions into one by sequential application,
+    i.e. Sequential(f, g)(x) = (g ∘ f)(x) = g(f(x)).
+
+    Can either compose ``map_seq``-style single-segment processor functions or ``map_seq_stream``-style
+    iterators operating on multiple segments. Just make sure not to mix both styles.
+    """
+
+    def __init__(self, *postprocessing_funcs: Callable[[T], T]):
```

Comment:
```suggestion
    def __init__(self, *postprocessing_funcs: Callable):
```
As discussed. Also, see the warning you currently get.

Reply:
ah, hm, that's another way.